drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -839,47 +839,6 @@ static void nrf5_config_mac_keys(struct ieee802154_key *mac_keys)
 }
 #endif /* CONFIG_IEEE802154_2015 */
 
-#if defined(CONFIG_IEEE802154_CSL_ENDPOINT)
-static void nrf5_receive_at(uint32_t start, uint32_t duration, uint8_t channel, uint32_t id)
-{
-	/*
-	 * Workaround until OpenThread (the only CSL user in Zephyr so far) is able to schedule
-	 * RX windows using 64-bit time.
-	 */
-	uint64_t rx_time = target_time_convert_to_64_bits(start);
-
-	nrf_802154_receive_at(rx_time, duration, channel, id);
-}
-
-static void nrf5_config_csl_period(uint16_t period)
-{
-	nrf_802154_csl_writer_period_set(period);
-
-	/* Update the CSL anchor time to match the nearest requested CSL window, so that
-	 * the proper CSL Phase in the transmitted CSL Information Elements can be injected.
-	 */
-	if (period > 0) {
```

Comment:
This check is missing in the new code. Are we sure it's not needed?

Reply:
in `sub_mac.cpp`  `HandleCslTimer()` is always called after `IgnoreError(Get<Radio>().EnableCsl()` and only if `(mCslPeriod > 0) `, where `mCslPeriod = aPeriod` which is passed to `EnableCsl` .  in  `HandleCslTimer()`, `Get<Radio>().UpdateCslSampleTime` and in new implementation it leads to `nrf_802154_csl_writer_anchor_time_set(nrf5_data.csl_rx_time)` call. So if I'm correct- in situation when `aPeriod=0`, `nrf_802154_csl_writer_anchor_time_set()` won't be called